docs/usage/user-stories/swissquote.md

```diff
@@ -0,0 +1,191 @@
+
+### Update your dependencies regularly
+
+This one sounds easy, every few weeks you check what’s outdated, bump the versions and run your CI. If it succeeds, you push the change. Maybe there is even a way to automate that ?
+
+#### Let's give Renovate a try
+
+One day in november 2019, I discovered that [Renovate](https://www.mend.io/free-developer-tools/renovate/) provides a Docker image that you can run on premise with your own package registries. The app creates Pull Requests automatically to inform you of dependencies updates and the CI can then build it automatically so you know if it’s safe to merge or not.
+
+On our first try, we enabled 30 repositories, a cron task was running every hour to create Pull Requests.
+
+We received 700 Pull Requests in the first month, it was a never ending Pull Request Whack-a-mole every time we merged one, another replaced it.
+
+![A cat playing Whack-a-mole with a finger](./swissquote_cat_whack_a_mole.gif)
+Me and my team merging Pull Requests
+
+The awesome thing with Renovate is that it’s very configurable, and this configuration can be shared. Very early on we created a shared configuration for our team with some custom policies, here are a few things we decided to do
+
+- Group PRs for minor and patch dependencies
+- Internal dependencies could create a PR anytime of the day
+- Third party dependencies could create PRs only during the weekend
+
+This helped a lot to reduce the noise in PRs, the second month we got 400 Pull Requests, and on the third month only 200.
+
+## What we learned from automating dependencies updates
+
+- You need to be confident that your code coverage will warn you in case of updates. At the beginning we missed quite a few breaking updates because the build was green but the application broke as soon as it was deployed.
+- Once you’re confident enough, auto-merge is a must have. In our team we enabled Renovate on ~100 of our repositories and generally spend 1–2 hours per week to stay up-to-date.
+- “On the bleeding edge it’s not the edge that’s bleeding; it’s you”. When updating to a new major version as soon as it’s released you might encounter some surprises. It happened to us a few times that a patch version breaks the library. Usually a fix came out the next day, but we still spent a few hours debugging why the update broke our applications. We’ve opened quite a few Issues and sent some PRs to fix issues like this.
+- Updating dependencies is one thing, but when should you release them ? As our team mostly provides libraries, we don’t want to release them on every dependency upgrade (as this would create PRs downstream and create noise for them). We’ve decided to release right after critical upgrades or contributions and a dashboard informs us when a repository hasn’t been released for 30 days.
```

Comment:
```suggestion
- You need to be confident that your code coverage will warn you in case of updates. At the beginning we missed quite a few breaking updates because the build was green but the application broke as soon as it was deployed.
- Once you’re confident enough, auto-merge is a must have. In our team we enabled Renovate on ~100 of our repositories and generally spend 1–2 hours per week to stay up-to-date
- “On the bleeding edge it’s not the edge that’s bleeding; it’s you”. When updating to a new major version as soon as it’s released you might encounter some surprises. It happened to us a few times that a patch version breaks the library. Usually a fix came out the next day, but we still spent a few hours debugging why the update broke our applications. We’ve opened quite a few Issues and sent some PRs to fix issues like this
- Updating dependencies is one thing, but when should you release them ? As our team mostly provides libraries, we don’t want to release them on every dependency upgrade (as this would create PRs downstream and create noise for them). We’ve decided to release right after critical upgrades or contributions and a dashboard informs us when a repository hasn’t been released for 30 days
```

- Remove punctuation at list item end